va/http.go

```diff
@@ -75,27 +75,33 @@ func newHTTPClient(checkRedirect redirectChecker) http.Client {
 
 // httpValidationURL constructs a URL for the given IP address, path and port
 // combination. The port is omitted from the URL if it is the default HTTP
-// port. It is only used for constructing initial HTTP-01 validation requests
-// and so does not need to support constructing a URL with an HTTPS scheme.
-func httpValidationURL(validationIP net.IP, path string, port int) *url.URL {
+// port or the default HTTPS port. The protocol scheme of the URL is HTTP unless
+// useHTTPS is true. UseHTTPS should only be true when constructing validation
```

Comment:
Rather than a `useHTTPS bool`, why not a `scheme string`? And we could organize the arguments by URL order: scheme, validationIP, port, path.

Reply:
Alright, sounds reasonable.